### PR TITLE
Trinity.Core: Fix assembly blacklist in AssemblyPath

### PR DESCRIPTION
**Pull request by leasunhy**
Discussion: shall we blacklist assemblies from test frameworks (e.g. xunit, nunit, ...) ?

### CODE REVIEW

File: src/Trinity.Core/Utilities/MISC/AssemblyPath.cs [hunk 1]
```diff
@@ -58,9 +58,8 @@ internal static string MyAssemblyPath
                             if (asm == trinity_asm) continue;
                             if (asm.IsDynamic) continue;
                             if (type == typeof(System.RuntimeMethodHandle)) continue;
-                            if (type.FullName == "System.Linq.Enumerable+WhereSelectEnumerableIterator`2") continue;
-                            if (type.FullName == "System.Linq.Enumerable+WhereEnumerableIterator`1") continue;
-                            if (type.FullName == "System.Linq.Enumerable") continue;
+                            if (asm == typeof(object).Assembly) continue;
+                            if (asm == typeof(System.Linq.Enumerable).Assembly) continue;
```
Comment (yatli) [inline src/Trinity.Core/Utilities/MISC/AssemblyPath.cs:62]: Are these two assemblies enough to fix the test framework errors?

**Comment by yatli** [comment]
@leasunhy nope, as long as the test frameworks do not call into Trinity directly.

File: src/Trinity.Core/Utilities/MISC/AssemblyPath.cs (hunk 1 above)
Comment (leasunhy) [reply src/Trinity.Core/Utilities/MISC/AssemblyPath.cs:62]: Yes, when the test fails, the `type` was `System.RuntimeMethodHandle` and its containing assembly was `mscorlib`, which is where `object` is defined.

Comment (leasunhy) [reply src/Trinity.Core/Utilities/MISC/AssemblyPath.cs:62]: Not tested for coreclr yet.

Comment (yatli) [reply src/Trinity.Core/Utilities/MISC/AssemblyPath.cs:62]: @leasunhy I see.

**Review by yatli — APPROVED** [review]

**Pull request merged**